src/hermit.js

```diff
@@ -240,169 +98,105 @@ export class HermitExportExtendedPublicKey extends HermitReader {
-    const {xpub, pubkey} = result;
-    const bip32Path = result.bip32_path;
-
-    if (!xpub) {
-      if (pubkey) {
-        throw new Error("Make sure you export an extended public key and NOT a plain public key.");
-      } else {
-        throw new Error("No extended public key in QR code.");
-      }
+  parse(descriptorHex) {
+    if (!descriptorHex) {
+      throw new Error("No descriptor received from Hermit.");
     }
-    if (!bip32Path) {
-      throw new Error("No BIP32 path in QR code.");
+    const descriptor = Buffer.from(descriptorHex, 'hex').toString('utf8');
+    const result = descriptor.match(DESCRIPTOR_REGEXP);
+    if (result && result.length == 4) {
+      return {
+        rootFingerprint: result[1],
+        bip32Path: `m${result[2]}`,
+        xpub: result[3],
+      };
+    } else {
+      throw new Error("Invalid descriptor received from Hermit.");
     }
-    result.bip32Path = bip32Path;
-    Reflect.deleteProperty(result, "bip32_path");
-    return result;
   }
-
 }
 
 /**
- * Returns signature request data to display in a QR code for Hermit
- * and reads the signature data passed back by Hermit in another QR
- * code.
+ * Displays a signature request for Hermit's `sign` command and reads
+ * the resulting signature.
  *
- * NOTE: Transactions with inputs & outputs to non-P2SH addresses are not supported by Hermit.
+ * This interaction class works in tandem with the `BCURDecoder`
+ * class.  The `BCURDecoder` parses data from Hermit, this class
+ * interprets it.
  *
- * @extends {module:hermit.HermitDisplayer}
+ * @extends {module:hermit.HermitInteraction}
  * @example
- * const interaction = new HermitSignTransaction({inputs, outputs, bip32Paths});
- * console.log(interaction.request());
- * // "IJQXGZI..."
+ * const interaction = new HermitSignMultisigTransaction({psbt});
+ * const urParts = interaction.request();
+ * console.log(urParts);
+ * // [ "ur:...", "ur:...", ... ]
  *
- * // Display a QR code containing the above data to Hermit running
- * // `sign-bitcoin` and it will return another QR code which needs
- * // parsed.
- * const encodedString = readHermitQRCode(); // application dependent
- * const signatures = interaction.parse(encoodedString);
- * console.log(signatures);
- * // ["ababa...01", ... ]
+ * displayQRCodeSequence(urParts);
+ * // ...
+ *
+ * const data = readQRCodeSequence();  // using BCURDecoder
+ * const signedPSBT = interaction.parse(data);
+ * console.log(signedPSBT);
+ * // "cHNidP8B..."
  *
  */
-export class HermitSignTransaction extends HermitDisplayer {
+export class HermitSignMultisigTransaction extends HermitInteraction {
 
   /**
    *
    * @param {object} options - options argument
-   * @param {array<object>} options.inputs - inputs for the transaction
-   * @param {array<object>} options.outputs - outputs for the transaction
-   * @param {array<string>} options.bip32Paths - BIP32 paths
+   * @param {array<object>} options.psbt - unsigned PSBT to sign
```

Comment:
add doc for returnSignatureArraay

`@param {bool} options.returnSignatureArray - return a signed PSBT or an array of signatures (useful in Caravan's testing app)`